internal/command/deploy/machines_tigrisstatics.go

```diff
@@ -0,0 +1,617 @@
+				//                available again. Maybe we need another solution, such as storing the bucket name in the app's metadata.
+				return flyerr.GenericErr{
+					Err:      fmt.Sprintf("tigris bucket '%s' already exists in another organization", md.tigrisStatics.bucket),
+					Descript: "",
+					Suggest:  fmt.Sprintf("try deleting this tigris bucket (`fly storage destroy %s`), waiting a few moments, then attempt to deploy again", md.tigrisStatics.bucket),
+					DocUrl:   "",
+				}
+			}
+		}
+	}
+
+	org, err := client.GetOrganizationBySlug(ctx, md.app.Organization.Slug)
+	if err != nil {
+		return err
+	}
+
+	params := extensions.ExtensionParams{
+		Organization:         org,
+		Provider:             "tigris",
+		Options:              gql.AddOnOptions{},
+		ErrorCaptureCallback: nil,
+		OverrideRegion:       md.appConfig.PrimaryRegion,
+		OverrideName:         &md.tigrisStatics.bucket,
+	}
+	params.Options["website"] = map[string]interface{}{
+		"domain_name": "",
+	}
+	params.Options["accelerate"] = false
+	// TODO(allison): Make sure we still need this when virtual services drop :)
+	params.Options["public"] = true
+
+	ext, err := extensions.ProvisionExtension(ctx, params)
+	if err != nil {
+		return err
+	}
+
+	secrets := ext.Data.Environment.(map[string]interface{})
+
+	tokenizedKey, err := md.staticsTokenizeTigrisSecrets(ctx, org, secrets, tokenizerSealKey)
+
+	// TODO(allison): Temporary, while working on the POC for Tokenizer
+	return os.WriteFile("tokenized_key", ([]byte)(tokenizedKey), 0644)
+
+	return err
+}
+
+func (md *machineDeployment) staticsTokenizeTigrisSecrets(
+	ctx context.Context,
+	org *fly.Organization,
+	secrets map[string]interface{},
+	tokenizerSealKey string,
+) (string, error) {
+
+	client := flyutil.ClientFromContext(ctx)
+
+	orgId, err := strconv.ParseUint(org.InternalNumericID, 10, 64)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode org ID for %s: %w", org.Slug, err)
+	}
+
+	// TODO(allison): We should just pass the whole app to deploy instead of re-grabbing it here
+	app, err := client.GetApp(ctx, md.app.Name)
+	if err != nil {
+		return "", err
+	}
+	appId := uint64(app.InternalNumericID)
+
+	type sigv4Processor struct {
+		AccessKey string `json:"access_key"`
+		SecretKey string `json:"secret_key"`
+	}
+	type flyioMacaroonAuthConfig struct {
+		Access flyio.Access `json:"access"`
+	}
+	type tokenizeInput struct {
+		Sigv4Processor          sigv4Processor          `json:"sigv4_processor"`
+		FlyioMacaroonAuthConfig flyioMacaroonAuthConfig `json:"flyio_macaroon_auth"`
+		AllowedHosts            []string                `json:"allowed_hosts"`
+	}
+
+	// TODO(allison): How do we handle moving an app between orgs?
+	//                We're locking this token behind a hard dependency on the App ID and Org ID, but I don't believe
+	//                *either* of these stay static when moving from one org to another.
```

Comment:
Yeah. The org id would change, but the app id would remain the same. The sealed token would need to be regenerated in this case.